src/js/component.js

```diff
@@ -524,6 +525,57 @@ class Component {
     return currentChild;
   }
 
+  /**
+   * Adds an SVG icon element to another element or component.
+   *
+   * @param {string} iconName
+   *        The name of icon. A list of all the icon names can be found at 'sandbox/svg-icons.html'
+   *
+   * @param {Element} [el=this.el()]
+   *        Component or element to set the title on. Defaults to the current Component's element.
+   *
+   * @return {Element}
+   *        The newly created icon element.
+   */
+  setIcon(iconName, el = this.el()) {
+    // TODO: In v9 of video.js, we will want to remove font icons entirely.
+    // This means this check, as well as the others throughout the code, and
+    // the unecessary CSS for font icons, will need to be removed.
+    // See https://github.com/videojs/video.js/pull/8260 as to which components
+    // need updating.
+    if (!this.player_.options_.useSVGIcons) {
+      return;
+    }
+
+    const xmlnsURL = 'http://www.w3.org/2000/svg';
+
+    // The below creates an element in the format of:
+    // <span><svg><use>....</use></svg></span>
+    const iconContainer = Dom.createEl('span', {
+      className: 'vjs-icon-placeholder vjs-svg-icon'
+    }, {'aria-hidden': 'true'});
+
+    const svgEl = document.createElementNS(xmlnsURL, 'svg');
+
+    svgEl.setAttributeNS(null, 'viewBox', '0 0 512 512');
+    const useEl = document.createElementNS(xmlnsURL, 'use');
+
+    svgEl.appendChild(useEl);
+    useEl.setAttributeNS(null, 'href', `../images/icons.svg#vjs-icon-${iconName}`);
```

Comment:
I know we talked about this path being sufficient, but the more I think about it, the more it makes sense for there to be an additional Video.js option for the URL for the SVG sprite.

It could be `../images/icons.svg` by default and that should work for the Video.js CDN or unpkg or similar, but I can see use-cases where a Video.js user will want to point that to a different location.

This would also allow people to create their own icons.

```js
const player = videojs('example', {
  iconsUrl: '//my-cdn.com/my-icons.svg'
});
```

Reply:
This is a very good idea! My only question is, do we expect these custom SVG sprites to have the exact same definitions of SVGs as our SVG sprite?

For icons that already exist in VJS, we expect them to have a certain id on the SVG element. If the custom sprite were to work with this implementation, it would need to have those same ids on the `defs` in their sprite, or else the preexisting icons won't show up.

Would it make more sense to be able to just pass that URL into the `setIcon` function as a param? This way, if a user wanted to overwrite an icon, they could get whatever component they want and call the function to update the icon from their sprite url. This would also allow them to avoid sprites entirely and call a file with a single SVG.

Definitely open to all ideas!